docs/canonical-ways-to-investigate-bugs.md

```diff
@@ -0,0 +1,368 @@
+```
+{
+  status: '{"errors":{"amount":"{\\"name\\":\\"NotEnoughBalance\\",\\"message\\":\\"NotEnoughBalance\\",\\"stack\\":\\"Error\\\\n    at new CustomError (/Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/errors/lib/helpers.js:27:18)\\\\n    at remapLibcoreErrors (/Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/live-common/lib/libcore/platforms/nodejs.js:543:16)\\\\n    at remapLibcoreErrors (/Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/live-common/lib/libcore/errors.js:18:10)\\\\n    at /Users/grenaudeau/dev/ledger-live-common/cli/node_modules/@ledgerhq/live-common/lib/libcore/getFeesForTransaction.js:46:42\\"}"},"warnings":{},"estimatedFees":"0","amount":"10000000000","totalSpent":"10000000000"}',
+  transaction: '{"amount":"10000000000","recipient":"znaE1JH6YjjWiG23bqdCkBiC4KwZ9vo7hB6","useAllAmount":false,"family":"bitcoin","feePerByte":"1","networkInfo":{"family":"bitcoin","feeItems":{"items":[{"key":"0","speed":"high","feePerByte":"10"},{"key":"1","speed":"standard","feePerByte":"10"},{"key":"2","speed":"low","feePerByte":"10"}],"defaultFeePerByte":"10"}}}'
+}
+```
+
+But if all is good, I will have some data:
+
+```
+{
+  status: '{"errors":{},"warnings":{},"estimatedFees":"226","amount":"10000000","totalSpent":"10000226"}',
+  transaction: '{"amount":"10000000","recipient":"znaE1JH6YjjWiG23bqdCkBiC4KwZ9vo7hB6","useAllAmount":false,"family":"bitcoin","feePerByte":"1","networkInfo":{"family":"bitcoin","feeItems":{"items":[{"key":"0","speed":"high","feePerByte":"10"},{"key":"1","speed":"standard","feePerByte":"10"},{"key":"2","speed":"low","feePerByte":"10"}],"defaultFeePerByte":"10"}}}'
+}
+```
+
+Just using `getTransactionStatus` will show you if a problem occur typically related to fetching fees or any other problem related to building a transaction. Now if the problem occurs at the device time, this will not be enough...
+
+### A problem during send flow, _during the device step_
+
+Complex user's problem can appear during the Send flow. It often occur that the problem is specific to the user's Account and that we don't reproduce on our side.
+
+In such case, the same technique can also be used by replaying APDUs. **Make sure to use `--disable-broadcast` to not broadcast the transaction.**
+
+The technique involve both the sync and the replay logic. We need to know beforehand what the xpub because the transaction logic starts by having a synchronized account.
+
+Essentially we can do this:
+
+```
+ DEVICE_PROXY_URL=ws://localhost:8435 ledger-live send --disable-broadcast -c ZEN --xpub xpub6DBXGC1faXnWSWofjAXjfxihNmYEcJj7KcWytuc7AZK39zS8KdgbosCjnPyHRbwKxcJxnWbzLwZYaPHFE3zoQZmGDbPzwRuoSfFaeSTVwUT --recipient znaE1JH6YjjWiG23bqdCkBiC4KwZ9vo7hB6 --amount 0.1
+```
+
+By using the exact same parameter as the user we could be able to reproduce, even tho in practice we will likely fall into problems because transactions tends to not be identically reproductible / deterministic.
```

Comment:
```suggestion
By using the exact same parameter as the user we could be able to reproduce, even though in practice we will likely fall into problems because transactions tends to not be identically reproductible / deterministic.
```